prymer/api/primer_pair.py

```diff
@@ -78,8 +74,28 @@ class PrimerPair(OligoLike):
     right_primer: Oligo
     amplicon_tm: float
     penalty: float
+    name: Optional[str] = None
     amplicon_sequence: Optional[str] = None
 
+    def __post_init__(self) -> None:
+        # Force generation of the amplicon span, which validates the relative positions
+        # of the left and right primers
+        amp = self.amplicon
+
+        # If supplied, bases must be the same length as the span length
```

Comment:
ditto, do we think that the amplicon supplied here matches the genome (i.e. the span) or can the amplicon be an alternative haplotype, and span is just the mapping relative to the reference (and not relative to the actual haplotype being amplified).  I suppose one could suggest that the _mapping_ should be relative to the haplotype instead of the genome.

Reply:
Yeah - I think until we or someone else tries to make a designer that targets a haplotype other than the one in the FASTA file, we should leave these checks in place (they were there previously, I just simplified).